src/@types/CommandContext.d.ts

```diff
@@ -0,0 +1,103 @@
+import type {
+	Client,
+	CommandInteraction,
+	CommandInteractionOption,
+	DMChannel,
+	Guild,
+	GuildMember,
+	GuildTextBasedChannel,
+	InteractionReplyOptions,
+	Message,
+	MessageReplyOptions,
+	User,
+} from 'discord.js';
+
+declare global {
+	/** Information relevant to a command invocation. */
+	interface CommandContext {
+		/** The command invocation interaction. */
+		readonly interaction: CommandInteraction;
+
+		/** Our own signed-in Discord client. */
+		readonly client: Client<true>;
+
+		/** The place to submit informative debug messages. */
+		readonly logger: Console; // TODO: More extensible logger than `console`?
+
+		/** The guild in which the command was invoked. */
+		readonly guild: Guild | null;
+
+		/** The channel in which the command was invoked. */
+		readonly channel: GuildTextBasedChannel | DMChannel | null;
+
+		/** The user who invoked the command. */
+		readonly user: User;
+
+		/** The guild member who invoked the command. */
+		readonly member: GuildMember | null;
+
+		/** The UNIX time at which the command was invoked. */
+		readonly createdTimestamp: number;
+
+		/** The options that were given to the command. */
+		readonly options: ReadonlyArray<CommandInteractionOption<'cached'>>; // TODO: Make this a generic tuple
+
+		/** Instructs Discord to keep interaction handles open long enough for long-running tasks to complete. */
+		prepareForLongRunningTasks: (ephemeral?: boolean) => void | Promise<void>;
+
+		/** Sends a typing indicator that ends in about 10 seconds or when a new message is sent. */
+		sendTyping: () => void;
+
+		/**
+		 * Sends a DM or ephemeral reply to the command's sender.
+		 *
+		 * In the case of an interaction that was publicly deferred (e.g.
+		 * using `prepareForLongRunningTasks(false)`), this function will
+		 * edit the public reply. The message sent here will then be public.
+		 *
+		 * @param options The message payload to send.
+		 * @param viaDM Whether we should reply in DMs.
+		 */
+		replyPrivately: (
+			options:
+				| string //
+				| Omit<MessageReplyOptions, 'flags'>
+				| Omit<InteractionReplyOptions, 'flags'>,
+			viaDM?: true
+		) => Promise<void>;
+
+		/** Replies to the command invocation, optionally pinging the command's sender. */
+		reply: (
+			options:
+				| string
+				| Omit<MessageReplyOptions, 'flags'>
+				| (Omit<InteractionReplyOptions, 'flags'> & {
+						shouldMention?: boolean;
+				  })
+		) => Promise<void>;
```

Comment:
From my experience, discord.js doesn't allow you to specify whether to send a ping when replying to a message (like users can). Has that changed?

Reply:
That has! See the [`allowedMentions`](https://discordjs.guide/additional-info/changes-in-v13.html#allowed-mentions) object. That object has a property, `replied_user`, that determines whether a message should ping the user whose message is being replied to, in much the same way as toggling the "@" button. When `allowedMentions` is given, `replied_user` [defaults to `false`](https://discord.com/developers/docs/resources/channel#allowed-mentions-object), barring a bug in discord.js or inaccurate documentation.